intermediate_source/rpc_param_server_tutorial.rst

```diff
@@ -0,0 +1,398 @@
+
+Implementing a Parameter Server Using Distributed RPC Framework
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Author**\ : `Rohan Varma <https://github.com/rohan-varma>`_
+
+This tutorials walks through a simple example of implementing a [parameter server] using PyTorch's `Distributed RPC framework <https://pytorch.org/docs/stable/rpc.html>`__. This is a framework in which a set of servers store parameters, such as large embedding tables, and several trainers query the parameter servers in order to retrieve the most up to date parameters. These trainers can run a training loop locally and ocassionally synchronize with the parameter server to get the latest parameters. For more reading on the parameter server approach, check out `this paper <https://www.cs.cmu.edu/~muli/file/parameter_server_osdi14.pdf>`_.
+
+Using the Distributed RPC Framework, we'll build a training setup where multiple machines can communicate tensors over RPC and remotely train a model, fetch model parameters, or obtain gradients through the use of remote handles given by  the `RRef <https://pytorch.org/docs/stable/rpc.html#torch.distributed.rpc.RRef>`_ API. Further, these machines will all participate in the backwards pass through stitching of the autograd graph across multiple nodes using distributed autograd.
```

Comment:
Further, these machines will all participate in the backwards pass through stitching of the autograd graph across multiple nodes using distributed autograd.

->

Each trainer will launch its dedicated backward pass in a distributed fashion through stitching of the autograd graph across multiple nodes using distributed autograd.